drivers/crypto/crypto_npcx_sha.c

```diff
@@ -0,0 +1,220 @@
+	case CRYPTO_HASH_ALGO_SHA384:
+		sha_type = NCL_SHA_TYPE_2_384;
+		break;
+	case CRYPTO_HASH_ALGO_SHA512:
+		sha_type = NCL_SHA_TYPE_2_512;
+		break;
+	default:
+		LOG_ERR("Unexpected algo: %d", npcx_session->algo);
+		return -EINVAL;
+	}
+
+	if (!ctx->started) {
+		ret = NPCX_NCL_SHA->start(npcx_ctx->handle, sha_type);
+		if (ret != NCL_STATUS_OK) {
+			LOG_ERR("Could not compute the hash, err:%d", ret);
+			return -EINVAL;
+		}
+		ctx->started = true;
+	}
+
+	if (pkt->in_len != 0) {
+		ret = NPCX_NCL_SHA->update(npcx_ctx->handle, pkt->in_buf, pkt->in_len);
+		if (ret != NCL_STATUS_OK) {
+			LOG_ERR("Could not update the hash, err:%d", ret);
+			ctx->started = false;
+			return -EINVAL;
+		}
+	}
+
+	if (finish) {
+		ctx->started = false;
+		ret = NPCX_NCL_SHA->finish(npcx_ctx->handle, pkt->out_buf);
+		if (ret != NCL_STATUS_OK) {
+			LOG_ERR("Could not compute the hash, err:%d", ret);
+			return -EINVAL;
+		}
+	}
+
+	return 0;
+}
+
+static int npcx_hash_session_setup(const struct device *dev, struct hash_ctx *ctx,
+				   enum hash_algo algo)
+{
+	int ctx_idx;
+	struct npcx_sha_context *npcx_ctx;
+
+	if (ctx->flags & ~(NPCX_HASH_CAPS_SUPPORT)) {
+		LOG_ERR("Unsupported flag");
+		return -EINVAL;
+	}
+
+	if ((algo != CRYPTO_HASH_ALGO_SHA256) && (algo != CRYPTO_HASH_ALGO_SHA384) &&
+	    (algo != CRYPTO_HASH_ALGO_SHA512)) {
+		LOG_ERR("Unsupported algo: %d", algo);
+		return -EINVAL;
+	}
+
+	ctx_idx = npcx_get_unused_session_index();
+	if (ctx_idx < 0) {
+		LOG_ERR("No free session for now");
+		return -ENOSPC;
+	}
+
+	npcx_sessions[ctx_idx].algo = algo;
+
+	ctx->drv_sessn_state = &npcx_sessions[ctx_idx];
+	ctx->started = false;
+	ctx->hash_hndlr = npcx_sha_compute;
+
+	npcx_ctx = &npcx_sessions[ctx_idx].npcx_sha_ctx;
+	NPCX_NCL_SHA->init_context(npcx_ctx->handle);
+	NPCX_NCL_SHA->power(npcx_ctx->handle, 1);
+	NPCX_NCL_SHA->init(npcx_ctx->handle);
+	NPCX_NCL_SHA->reset(npcx_ctx->handle);
+
+	return 0;
+}
+
+static int npcx_hash_session_free(const struct device *dev, struct hash_ctx *ctx)
+{
+	struct npcx_sha_session *npcx_session = (struct npcx_sha_session *)ctx->drv_sessn_state;
+	struct npcx_sha_context *npcx_ctx = &npcx_session->npcx_sha_ctx;
+
+	NPCX_NCL_SHA->reset(npcx_ctx->handle);
+	NPCX_NCL_SHA->power(npcx_ctx->handle, 0);
+	NPCX_NCL_SHA->finalize_context(npcx_ctx->handle);
+	npcx_session->in_use = false;
+
+	return 0;
+}
+
+static int npcx_query_caps(const struct device *dev)
+{
+	return NPCX_HASH_CAPS_SUPPORT;
+}
+
+static int sha_npcx_init(const struct device *dev)
+{
+	return 0;
+}
+
+static struct crypto_driver_api npcx_crypto_api = {
+	.hash_begin_session = npcx_hash_session_setup,
+	.hash_free_session = npcx_hash_session_free,
+	.query_hw_caps = npcx_query_caps,
+};
+
+DEVICE_DT_INST_DEFINE(0, &sha_npcx_init, NULL, NULL, NULL, POST_KERNEL, CONFIG_CRYPTO_INIT_PRIORITY,
+		      &npcx_crypto_api);
+BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 1,
+	     "only one 'nuvoton_npcx_hsa' compatible node can be supported");
```

Comment:
Thanks for pointing this typo. I have fixed it in the new patch.

Reply:
It is strange because I fixed and push it with the other changes together.
I also see it modified in the web UI.
![image](https://user-images.githubusercontent.com/75654292/208058451-0562e694-8e7a-4044-a202-b90176093d20.png)

Reply:
Sorry what I meant was that the message should use the comma and dash, version (the one that goes in the dts file) rather than the underscore one (the one that goes in the code. Did not realize there was a typo in "sha" :-D